python/cudf/cudf/pandas/_wrappers/pandas.py

```diff
@@ -1711,6 +1713,22 @@ def holiday_calendar_factory_wrapper(*args, **kwargs):
     )
 
 
+def is_cudf_pandas_obj(obj):
+    return is_proxy_object(obj)
+
+
+def is_cudf_pandas_dataframe(obj):
+    return is_proxy_object(obj) and isinstance(obj, DataFrame)
+
+
+def is_cudf_pandas_series(obj):
+    return is_proxy_object(obj) and isinstance(obj, Series)
+
+
+def is_cudf_pandas_index(obj):
+    return is_proxy_object(obj) and isinstance(obj, Index)
```

Comment:
Are these functions actually useful? It seems like it would be better to just tell the user how to check this themselves with `is_proxy_object` + `isinstance`.

Reply:
Yes, we could do that but my intention here was to just have single API that informs the users. It will end up being a verbose pattern in libraries that operate `cudf`/`cudf.pandas` aware.

Reply:
My suggestion in a thread above was to instead implement a single method (bikeshed the name as much as you wish, I'm mostly concerned with suggesting an implementation)
```
def isinstance_cudf_pandas(obj, type):
    return is_proxy_object(obj) and isinstance(obj, type)
```

so that they don't need to be aware of `is_proxy_object` and it "feels like" they're using `isinstance`.